lib/listen/record.rb

```diff
@@ -75,31 +63,36 @@ def build
 
     private
 
+    # TODO: refactor/refactor out
+    def add_dir(dir, rel_path, children)
+      rel_path = '.' if [nil, '', '.'].include? rel_path
+      dirname, basename = Pathname(rel_path).split.map(&:to_s)
+      basename = '.' if [nil, '', '.'].include? basename
+      if [nil, '', '.'].include?(dirname) && basename == '.'
+        @paths[dir.to_s]['.'] = Hash[children.map { |child| [child, {} ] } ]
+      else
+        @paths[dir.to_s][rel_path] ||= {}
```

Comment:
As above I feel there's separation of thought between '.' and other directories that I don't understand why need treating differently.

Other folders are not populated here, they are populated as each entry is examined in _fast_try_file for example. So why does '.' need populating here.